core/shared/src/main/scala/fs2/concurrent/Queue.scala

```diff
@@ -454,6 +454,37 @@ trait InspectableQueue[F[_], A] extends Queue[F, A] {
   def getSize: F[Int]
 }
 
+/**
+  * Like [[InspectableQueue]], but allows allows signalling of no further enqueues by enqueueing `None`.
+  * Optimizes dequeue to minimum possible boxing.
+  */
+trait NoneTerminatedInspectableQueue[F[_], A] extends NoneTerminatedQueue[F, A] {
+
+  /**
+    * Returns the element which would be dequeued next,
+    * but without removing it. Completes when such an
+    * element is available.
+    */
+  def peek1: F[A]
+
+  /**
+    * The time-varying size of this `Queue`.
+    * Emits elements describing the current size of the queue.
+    * Offsetting enqueues and de-queues may not result in refreshes.
+    *
+    * Finally, note that operations like `dequeue` are optimized to
+    * work on chunks when possible, which will result in faster
+    * decreases in size that one might expect.
+    * More granular updates can be achieved by calling `dequeue1`
+    * repeatedly, but this is less efficient than dequeueing in
+    * batches.
+    */
+  def size: Stream[F, Int]
```

Comment:
Should we use a `Signal` for this?  _If I understand correctly_, the difference between a Signal and a Stream is that the latter has a returns a fixed number of elements, whereas the former can be sampled by the consumer downstreams and only get values at those points.